src/libstd/fs.rs

```diff
@@ -1338,10 +1338,10 @@ pub fn remove_dir_all<P: AsRef<Path>>(path: P) -> io::Result<()> {
 ///
 /// // one possible implementation of walking a directory only visiting files
 /// fn visit_dirs(dir: &Path, cb: &Fn(&DirEntry)) -> io::Result<()> {
-///     if try!(fs::metadata(dir)).is_dir() {
+///     if dir.is_dir() {
 ///         for entry in try!(fs::read_dir(dir)) {
 ///             let entry = try!(entry);
-///             if try!(fs::metadata(entry.path())).is_dir() {
+///             if try!(entry.file_type()).is_dir() {
```

Comment:
This is actually changing the meaning of this example because `file_type()` doesn't follow symlinks, perhaps this could change to calling `is_dir` on the return value of `path` to preserve the same behavior?